### ISSUE TITLE
Request for Assistance with Library Imports in Sirius 

### ISSUE THREAD

**Issue by MohsenGholamigit** [eclipse-sirius/sirius-desktop#464]
Hello and good day,

I would like to use some libraries in Sirius and import them into my project. However, even though these libraries are imported correctly, I am unable to get the output from the code that utilizes them.

If you have any solutions or guidance, I would greatly appreciate it. Thank you very much!

Best regards,

**Comment by MohsenGholamigit** [comment]
Thank you for answering me, for example I want to use an external Java library and also use Sirius services, I create a new function and create an object from the imported class and finally output I will return Now when I try to use this output in Sirius in a textbox, it shows a blank page (due to an error of course). while the returned value is a string number or whatever. However, when I use standard libraries, there is no such problem at all.

**Comment by MohsenGholamigit** [comment]
> Hi,
> 
> Sorry but your question is not precise enough for us to help.
> 
> What kind of libraries are you talking about? What do you mean _precisely_ by "import them into my project", and by "get the output from the code that utilizes them"?
> 
> If you are talking about Java libraries you use in your modeler definition project (e.g. inside Java services), this is more of a general Eclipse question and not related to Sirius itself.
> 
> If you are talking about models copied to/referenced from a modeling project, I don't understand what "get the output from the code that utilizes them" means.

> Thank you for answering me, for example I want to use an external Java library and also use Sirius services, I create a new function and create an object from the imported class and finally output I will return Now when I try to use this output in Sirius in a textbox, it shows a blank page (due to an error of course). while the returned value is a string number or whatever. However, when I use standard libraries, there is no such problem at all.

**Comment by MohsenGholamigit** [comment]
> Hi,
> 
> Sorry but your question is not precise enough for us to help.
> 
> What kind of libraries are you talking about? What do you mean _precisely_ by "import them into my project", and by "get the output from the code that utilizes them"?
> 
> If you are talking about Java libraries you use in your modeler definition project (e.g. inside Java services), this is more of a general Eclipse question and not related to Sirius itself.
> 
> If you are talking about models copied to/referenced from a modeling project, I don't understand what "get the output from the code that utilizes them" means.

I have imported a library and added it to the manifest file and the plugin file. I also import in the Java service. However when I return the output to Sirius it gives an error saying the library was not recognized. While in services it is known as library.

**Comment by pcdavid** [comment]
> I have imported a library and added it to the manifest file and the plugin file.

Not sure about my interpretation, but reading between the lines I'd say the way you "imported the library" is invalid for and Eclipse-based product like Sirius, which work differently from plain/classical Java projects in terms of dependencies/classpath settings. But again, you don't give actual technical details of the concrete operations you did, so that's just a guess.

**Comment by MohsenGholamigit** [comment]
> > I have imported a library and added it to the manifest file and the plugin file.
> 
> Not sure about my interpretation, but reading between the lines I'd say the way you "imported the library" is invalid for and Eclipse-based product like Sirius, which work differently from plain/classical Java projects in terms of dependencies/classpath settings. But again, you don't give actual technical details of the concrete operations you did, so that's just a guess.

Yes, I think the import settings of the libraries in Sirius are different and it seems that the problem is from here. Can you guide me in this regard?
Thank you

**Comment by pcdavid** [comment]
Sorry but this is not Sirius-specific. In these regards, Sirius is a pretty normal Eclipse plug-in project. Look for documentation/tutorials on how to develop Eclipse plug-ins (not just Java programs developed *with* Eclipse, but plug-ins designed to run *in* Eclipse). Maybe use [this page](https://help.eclipse.org/latest/index.jsp?nav=%2F2) as a starting point.

**Comment by MohsenGholamigit** [comment]
> Sorry but this is not Sirius-specific. In these regards, Sirius is a pretty normal Eclipse plug-in project. Look for documentation/tutorials on how to develop Eclipse plug-ins (not just Java programs developed _with_ Eclipse, but plug-ins designed to run _in_ Eclipse). Maybe use [this page](https://help.eclipse.org/latest/index.jsp?nav=%2F2) as a starting point.

I hope this message finds you well. Thank you for your guidance. I was able to solve the issue with the help of the links you provided. However, I am now looking for advice and references on how to establish communication between the Sirius plugin and an external Eclipse plugin, so I can further develop my project. I would greatly appreciate it if you could recommend any API or documentation on this topic.

Thank you again.
